### PR TITLE
Cleanup: Remove server name stuff from NetVConnection

### PR DESCRIPTION
**Pull request by maskit**
Server name is `nullptr` unless it's indicated by TLS SNI extension. Probably not a something we want to have in `NetVConnection`.

### CODE REVIEW

**Comment by maskit** [comment]
http2_flow_control and txn_box_multi-ramp-1-3 failed.

**Comment by maskit** [comment]
[approve ci autest]

**Comment by maskit** [comment]
Same tests failed.

**Comment by maskit** [comment]
Tests previously failed were fixed but now remap_acl fails.

**Comment by maskit** [comment]
[approve ci autest]

File: src/api/InkAPI.cc [hunk 1]
```diff
@@ -7928,17 +7928,17 @@ TSVConnFdGet(TSVConn vconnp)
 const char *
 TSVConnSslSniGet(TSVConn sslp, int *length)
 {
-  char const     *server_name = nullptr;
-  NetVConnection *vc          = reinterpret_cast<NetVConnection *>(sslp);
-
-  if (vc == nullptr) {
+  if (sslp == nullptr) {
     return nullptr;
```
Comment (bryancall) [inline src/api/InkAPI.cc:7932]: I think length should also be set to 0 if we are returning a nullptr.

**Review by bryancall — APPROVED** [review]

File: src/api/InkAPI.cc (hunk 1 above)
Comment (maskit) [reply src/api/InkAPI.cc:7932]: I'll make that change separately with documentation update.

**Pull request merged**